utils/src/main/java/uk/gov/service/payments/commons/utils/prometheus/EcsTaskMetadataVersion4.java

```diff
@@ -0,0 +1,77 @@
+package uk.gov.service.payments.commons.utils.prometheus;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Optional;
+
+/**
+ * Extracts relevant information from the <a href="https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint-v4.html">AWS ECS task metadata endpoint version 4</a>.
+ */
+public class EcsTaskMetadataVersion4 {
+    
+    private final JSONObject ecsTaskMetadataJson;
+    private final String clusterName;
+    private final String accountName;
+    private final String ecsTaskId;
+    private final String containerImageTag;
+    private final String serviceName;
+    private final String instanceLabel;
+
+    /**
+     * Will throw a runtime exception if JSON is not valid.
+     * @param ecsTaskMetadataJson A valid <a href="https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint-v4.html#task-metadata-endpoint-v4-response">Task metadata JSON</a>
+     */
+    public EcsTaskMetadataVersion4(String ecsTaskMetadataJson) {
+        this.ecsTaskMetadataJson = new JSONObject(ecsTaskMetadataJson);
+        
+        String cluster = this.ecsTaskMetadataJson.getString("Cluster");
+        this.clusterName = cluster.substring(cluster.lastIndexOf("/") + 1);
+
+        this.accountName = clusterName.substring(0, clusterName.indexOf("-"));
+        
+        String taskARN = this.ecsTaskMetadataJson.getString("TaskARN");
+        this.ecsTaskId = taskARN.substring(taskARN.lastIndexOf("/") + 1);
+
+        JSONObject container = this.ecsTaskMetadataJson.getJSONArray("Containers").getJSONObject(0);
+        String image = container.getString("Image");
+        this.containerImageTag = image.substring(image.lastIndexOf(":") + 1);
+        
+        this.serviceName = container.getString("Name");
+
+        String instanceLabel;
+        try {
+            JSONArray networks = container.getJSONArray("Networks");
+            instanceLabel = (String) networks.getJSONObject(0).getJSONArray("IPv4Addresses").get(0);
```

Comment:
same question, presume you're wanting the address of the container collecting the metrics

Reply:
"IPv4Addresses" is a [list](https://github.com/alphagov/pay-java-commons/pull/691/files#diff-b2982d5a0905d509ec74e9247fe1e9bf0fa83a314d4732153b927a7ab0302212R41-R43), so just getting the first element here.

Reply:
yes but is the first element the correct value in all cases?

Reply:
I don't know enough to answer this question. The [doc](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint-v4.html#task-metadata-endpoint-v4-response) seems to suggest it is a singular IP address:
```
Networks
  The network information for the container, such as the network mode and IP address. This parameter is omitted if no network information is defined.
```

Reply:
So I'd suggest to leave this as is and get the first element for now.